### PR TITLE
fedora installer tests

### PR DESCRIPTION
**Pull request by emmettbutler**
This change adds single-step installation tests that run against the `Fedora-Cloud-Base-AmazonEC2.x86_64-39-20240912.0` AMI for the purpose of increasing the coverage of the test suite across different operating systems.

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/_context/virtual_machines.py [hunk 1]
```diff
@@ -331,3 +330,18 @@ def __init__(self, **kwargs) -> None:
             os_cpu="amd64",
             **kwargs,
         )
+
+
+class Fedora39amd64(_VirtualMachine):
+    def __init__(sef, **kwargs) -> None:
+        super().__init__(
+            "Fedora_39_amd64",
+            aws_config=_AWSConfig(ami_id="ami-0ba8f2988373d0735", ami_instance_type="t2.medium", user="fedora"),
```
Comment (robertomonteromiguel) [inline utils/_context/virtual_machines.py:339]: I think you are using the worng AMI id. See this message running the tests:
`06:21:01.786 INFO     Diagnostics:
06:21:01.786 INFO       aws:ec2:Instance (Fedora_39_amd64):
06:21:01.786 INFO         error: 1 error occurred:
06:21:01.786 INFO         	* creating EC2 Instance: OptInRequired: In order to use this AWS Marketplace product you need to accept terms and subscribe. To do so please visit https://aws.amazon.com/marketplace/pp?sku=ec8t8q98ms0b0idcq74tlco2a
06:21:01.786 INFO         	status code: 401, request id: 23227fff-a900-4042-8dfa-426554cecfce
06:21:01.786 INFO     
06:21:01.786 INFO       pulumi:pulumi:Stack (system-tests-vms-system-tests_onboarding):
06:21:01.786 INFO         error: update failed`

Pipeline: https://gitlab.ddbuild.io/DataDog/system-tests/-/pipelines/44748391

Try to find the correct ami from: aws console --> ec2 --> images --> amis --> search for public images

Comment (emmettbutler) [reply utils/_context/virtual_machines.py:339]: Every other AMI I use here gives me this message in the logs
```
command:remote:Command -Fedora_39_amd64-init-config creating (15s) Dial 1/inf failed: retrying
```

**Comment by emmettbutler** [comment]
@robertomonteromiguel I'm not sure what to do with the failing tests on this branch, since they pass on my local runs. Any tips?

**Comment by robertomonteromiguel** [comment]
> @robertomonteromiguel I'm not sure what to do with the failing tests on this branch, since they pass on my local runs. Any tips?

I run your pipeline forcing to update the AMIs.
That I can see is that you need to install docker compose.

**Comment by emmettbutler** [comment]
@robertomonteromiguel could you share the command you used to replicate that error?

**Comment by emmettbutler** [comment]
As far as I can tell, this test is now in a state where it's indicating a real incompatibility between Fedora 39 and auto-instrumentation.

**Comment by robertomonteromiguel** [comment]
> As far as I can tell, this test is now in a state where it's indicating a real incompatibility between Fedora 39 and auto-instrumentation.

I checked locally. 

- The agent is running and listening. No error traces on the agent.
- The app on docker container is running and the injection is performed.
- I see only this error:

```test-app  | {"level":"debug","ts":1727249824.710667,"caller":"preload_go/inject.go:126","msg":"results","envs":["DD_TRACING_ENABLED=true","DD_TAGS=_dd.injection.mode:docker","DD_RUNTIME_METRICS_ENABLED=true","DD_LOGS_INJECTION=true","DD_INJECTION_ENABLED=tracer","DD_TELEMETRY_FORWARDER_PATH=/opt/datadog-packages/datadog-apm-inject/0.20.0/inject/process"]}
test-app  | {"level":"debug","ts":1727249824.710783,"caller":"telemetry/client.go:189","msg":"request json","request":"{\"api_version\":\"v1\",\"request_type\":\"generate-metrics\",\"tracer_time\":1727249824,\"runtime_id\":\"741711ea-ec89-4542-a87d-9446bb9c3eb7\",\"seq_id\":1,\"application\":{\"service_name\":\"local\",\"env\":\"unknown\",\"tracer_version\":\"5.22.0\",\"language_name\":\"nodejs\",\"language_version\":\"unknown\"},\"host\":{\"hostname\":\"ee25ed8fa45f\",\"os\":\"Debian GNU/Linux\",\"os_version\":\"12 (bookworm)\",\"architecture\":\"amd64\"},\"payload\":{\"namespace\":\"tracers\",\"lib_language\":\"nodejs\",\"lib_version\":\"5.22.0\",\"series\":[{\"metric\":\"inject.success\",\"points\":[[1727249824,1]],\"type\":\"count\",\"tags\":[\"injector_version:0.20.0\",\"platform:docker\"],\"common\":true}]}}"}
test-app  | {"level":"error","ts":1727249824.7134311,"caller":"preload_go/process.go:239","msg":"error flushing telemetry","error":"Post \"http://localhost:8126/telemetry/proxy/api/v2/apmtelemetry\": dial tcp 127.0.0.1:8126: connect: connection refused","stacktrace":"github.com/datadog/preload_go.CommandLineEvaluator.FlushTelemetry\n\t/go/src/github.com/DataDog/auto_inject/preload_go/process.go:239\ngithub.com/datadog/preload_go.CommandLineEvaluator.Process.func1.1\n\t/go/src/github.com/DataDog/auto_inject/preload_go/process.go:306\ngithub.com/datadog/preload_go.InjectCmd.func1\n\t/go/src/github.com/DataDog/auto_inject/preload_go/inject.go:110\ngithub.com/datadog/preload_go.InjectCmd\n\t/go/src/github.com/DataDog/auto_inject/preload_go/inject.go:133\ngithub.com/datadog/preload_go.InjectFromParentProcess\n\t/go/src/github.com/DataDog/auto_inject/preload_go/inject.go:41\nmain.main\n\t/go/src/github.com/DataDog/auto_inject/preload_go/cmd/process/main.go:136\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271"}
test-app  | {"level":"debug","ts":1727249824.71361,"caller":"process/main.go:153","msg":"exiting process"}```

**My theory is that something in the configuration of the machine or in docker, makes that the containers cannot connect to the localhost (to the agent).
What do you think? how can we handle this?**

**Pull request closed without merging**